rlclientlib/model_mgmt/restapi_data_transport.cc

```diff
@@ -38,50 +41,62 @@ namespace reinforcement_learning { namespace model_management {
    * x-ms-version = 2017-04-17
    */
 
-  int restapi_data_transport::get_data_info(::utility::datetime& last_modified, ::utility::size64_t& sz, api_status* status) {
+  int restapi_data_transport::get_data_info(::utility::datetime& last_modified, ::utility::size64_t& sz, api_status* status, http_request request) {
 
-    // Build request URI and start the request.
-    auto request_task = _httpcli->request(methods::HEAD)
-      // Handle response headers arriving.
-      .then([&](http_response response) {
-      if ( response.status_code() != 200 )
-        RETURN_ERROR_ARG(_trace, status, http_bad_status_code, _httpcli->get_url());
+      // Get request URI and start the request.
+      request.headers() = _header;

```

Comment:
What does this line mean?

Reply:
get_data_info will be used in either cases of blob uri or end point. In case if it uses blob uri just normal header will be added where as in case if API end point is used header that has authentication details will be added